optimum/intel/utils/import_utils.py

```diff
@@ -196,8 +170,100 @@ def is_openvino_available():
     return _openvino_available
 
 
+def _create_pip_update_command(package_name: str, compatible_version: str, is_dev_version: bool = False) -> str:
+    return " ".join(
+        filter(
+            None,
+            (
+                "pip install",
+                "--pre" if is_dev_version else "",
+                "-U",
+                f"{package_name}=={compatible_version}",
+                "--extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly"
+                if is_dev_version
+                else "",
+            ),
+        )
+    )
+
+
+@functools.lru_cache(1)
 def is_openvino_tokenizers_available():
-    return _openvino_tokenizers_available
+    if not is_openvino_available():
+        return False
+
+    try:
+        import openvino_tokenizers
+    except ImportError:
+        logger.info(
+            "OpenVINO Tokenizers is not available, tokenizer won't be converted. To deploy models in production "
+            "with C++ code, please follow installation instructions: "
+            "https://github.com/openvinotoolkit/openvino_tokenizers?tab=readme-ov-file#installation\n"
+        )
+        return False
+
+    try:
+        pip_metadata_version = importlib_metadata.version("openvino")
+        is_nightly = False
+    except importlib_metadata.PackageNotFoundError:
+        try:
+            pip_metadata_version = importlib_metadata.version("openvino-nightly")
+            is_nightly = True
+        except importlib_metadata.PackageNotFoundError:
+            pip_metadata_version = False
+            is_nightly = False

```

Comment:
this should be done outside in my opinion (check if nightly)

Reply:
Do you mean in the separate function? It doesn't matter for other components, so I check it here.